clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java

```
@@ -79,6 +92,43 @@ public CredentialCache credentialCache() {
         return credentialCache;
     }
 
+    public double metricValue(String name) {
+        for (Map.Entry<MetricName, KafkaMetric> entry : metrics.metrics().entrySet()) {
+            if (entry.getKey().name().equals(name))
+                return entry.getValue().value();
+        }
+        fail("Metric not found, " + name + ", found=" + metrics.metrics().keySet());
+        return 0.0;
```

Nit: maybe throw an ordinary exception to avoid the need to return `0.0`.